### DISCUSSION TITLE
Common credentials across multiple postal Server - How to do that

### DISCUSSION THREAD

**Discussion by jaydeepnp** [postalserver/postal#2550] [Help with using Postal]
Hi ,

I am working on  setting 2 postal servers behind AWS NLB and for that I want to setup common credentials on both server. 
On both server below credentials is created with same password string so that it can be connected trough NLB in load balanced mode.

credentials to connect SMTP Server 
production@smtp.example.com 

Server1 with hostname --> smtp.example.com
Server2 with hostname  --> smtp01.example.com

Now when sendmail configured to relay all mails to 2nd server it gives error "SMTP DATA-2 protocol error: 530 From/Sender name is not valid"

Can someone please help how to achieve this requirement.

Regards,
Jaydeep

**Comment by willpower232** [comment]
The from address has to be a domain added to Postal. Did you share a database and rabbitmq between your two servers?

**Reply by jaydeepnp** [reply]
Will,

Thanks for your response, No both server independent postal setup with public IP and all DNS records for outbound email delivery.

Common SMTP authentication (using  exactly same credentials on both server) is required so that clients   (My Applications) can connect over ALB to submit emails.

Need help how we can achieve this.

Regards,
Jaydeep

**Reply by willpower232** [reply]
you will need to either share the database and rabbitmq so both servers have access to the same data and queue or manually duplicate all the settings between them

**Reply by jaydeepnp** [reply]
Will,

Thanks for your reply.

I was able to solve this problem  by setting  user_for_any =1 on db for root domain of the server in concern. 

As I have set both server as an independent postal SMTP server with their own DB and RabbitMQ I have configured SMTP credentials same on both server with exactly same password so that SMTP connection can  be authenticated when connection comes from ALB.

Both servers are behind ALB so that clients can only connect on  ALB for mail submission  and is getting submitted to any server in round robin  mode to have HA  of SMTP server.
 
Once again thanks for your help.

Regards,
Jaydeep
